contracts/bonding/libraries/SortedArrays.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "../../libraries/MathUtils.sol";
+
+import "@openzeppelin/contracts/utils/Arrays.sol";
+
+/**
+ * @title SortedArrays
+ * @dev Handles maintaining and looking up on sorted uint256 arrays.
+ */
+library SortedArrays {
```

Comment:
Is there any reusable, audited version of this? Seems like something that should be an openzeppelin contract to reduce surface area

Reply:
The reusable version of this is the `Arrays.findUpperBound`, which we do use internally for the more complex logic of actually implementing a binary search. Here we just adapt the "finde upper" to a "find lower" instead which is what we need for the bonding checkpoints. I might be making the interfaces even more similar right next, like making this function return just the index in the same way that OpenZeppelin's interface does it. Will make error checking less redundant in the checkpoint accessing logic.